Sim/SimPapas/src/PapasExportParticlesTool.cpp

```diff
@@ -18,13 +25,25 @@ StatusCode PapasExportParticlesTool::initialize() {
 
 StatusCode PapasExportParticlesTool::createOutputStructures() {
   m_particles = m_particlesHandle.createAndPut();
+  m_particleMCParticleAssociations = m_associationHandle.createAndPut();
   return StatusCode::SUCCESS;
 }
 
-StatusCode PapasExportParticlesTool::run(papas::Event& pevent) {
+StatusCode PapasExportParticlesTool::run(papas::Event& pevent, std::unordered_map<papas::Identifier, int>& links) {
   // convert each papas particle into an MCParticle
   std::string ptype = m_particleSubtype;
+  papas::HistoryHelper history(pevent);
+  
+  //put the original gen particles into a vector so they can be used to
+  //make the link association with the reconstructed particle
+  const fcc::MCParticleCollection* ptcs = m_iGenpHandle.get();
+  std::vector<fcc::ConstMCParticle> vecPtcs;
+  for (const auto& p : *ptcs) {
+     vecPtcs.push_back(p);
```

Comment:
but isn't ptcs already a vector? why do we need a second one?

Reply:
trying too hard... Thanks